distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -209,116 +210,66 @@ Text.characters self =
-Text.match self term mode=Matching_Mode.First matcher=Regex_Matcher.Value = case matcher of
-    _ : Text_Matcher ->
-        case_sensitivity = case matcher of
-            Text_Matcher.Case_Sensitive -> Case_Sensitivity.Sensitive
-            Text_Matcher.Case_Insensitive _ -> Case_Sensitivity.Insensitive
-        case self.locate term mode case_sensitivity of
-            Nothing -> Nothing
-            span -> span.text
-    _ : Regex_Matcher -> case mode of
-        Matching_Mode.First ->
-            case matcher.compile term . match self Matching_Mode.First of
-                Nothing -> Nothing
-                match -> match.span 0 . to_grapheme_span . text
-        Matching_Mode.Last ->
-            case matcher.compile term . match self Regex_Mode.All of
-                Nothing -> Nothing
-                matches -> matches.last.span 0 . to_grapheme_span . text
-
-## ALIAS find_all
-
-   Matches all occurrences text in `self` against the provided `term`, returning
-   a vector of matches.
+## Finds all the matches of the regular expression `pattern` in `self`,
+   returning a Vector. If not found, will be an empty Vector.
 
    Arguments:
-   - term: The pattern to match `self` against. We recommend using _raw text_
-     to write your patterns.
-   - matcher: If a `Text_Matcher`, the text is compared using case-sensitivity
-     rules specified in the matcher. If a `Regex_Matcher`, the term is used as a
-     regular expression and matched using the associated options.
+   - pattern: The pattern to match `self` against.
+   - case_sensitivity: Specifies if the text values should be compared case
+     sensitively.
 
    > Example
-     Find all substrings matching the regex.
-
-         example_match =
-             regex = "a[ab]c"
-             "aabcbbccaacaa".match regex == ["abc", "aac"]
-Text.match_all : Text -> (Text_Matcher | Regex_Matcher) -> Vector Text
-Text.match_all self term=".*" matcher=Regex_Matcher.Value = case matcher of
-    _ : Text_Matcher ->
-        case_sensitivity = case matcher of
-            Text_Matcher.Case_Sensitive -> Case_Sensitivity.Sensitive
-            Text_Matcher.Case_Insensitive _ -> Case_Sensitivity.Insensitive
-        self.locate_all term case_sensitivity . map .text
-    _ : Regex_Matcher ->
-        case matcher.compile term . match self Regex_Mode.All of
-            Nothing -> []
-            matches -> matches.map m-> m.span 0 . to_grapheme_span . text
+     Find the substring matching the regex.
+
+         example_find_all =
+              ## This matches `aabbbbc` @ character 0 and `abc` @ character 11
+             "aabbbbccccaabcaaaa".find_all "a[ab]+c"
+Text.find_all : Text -> Case_Sensitivity -> Vector Match ! Compile_Error
+Text.find_all self pattern=".*" case_sensitivity=Case_Sensitivity.Sensitive =
+    matcher = Regex_Matcher.Value case_sensitivity
+    case matcher.compile pattern . match self Regex_Mode.All of
```

Comment:
Lets go to `Regex.compile pattern` and then we can by pass the `Regex_Matcher`.

Reply:
Done; this requires bringing the defaulting logic into a helper in this file. Or should it perhaps go in Case_Sensitivity?

Reply:
My PR had the function in it - by all means grab from there.